src/viz/validation.ts

```diff
@@ -50,19 +50,19 @@ No files have been created.
 };
 
 export const parseBucketName = (bucketPath: string): string | undefined => {
-  const matches = bucketPath.match(/(gs:\/\/[^/\s]+)/);
+  const matches = bucketPath.match(/(gs:\/\/[-a-zA-Z]+)\/?([-a-zA-Z]+\/?)*$/);
```

Comment:
Simpler regex with no exponential blowup possibility:

```suggestion
  const matches = bucketPath.match(/^(?:gs:\/\/[\w-]+\/)?([\w-]+(?:\/[\w-]+)*)\/?$/);
```

Note that making this replacement will require you to change `matches[1]` back to `matches[0]`

Reply:
I did end up updating the regex based on the one you suggested here, and added a test to make sure it works with pretty large inputs.